client/components/command-palette/use-command-palette.tsx

```diff
@@ -79,24 +81,56 @@ export const useCommandPallette = ( {
 		( site ) => ! site.options?.is_domain_only
 	);
 
-	//Sort sites in the nested commands to be consistent with site switcher and /sites page
+	// Sort sites in the nested commands to be consistent with site switcher and /sites page
 	const { sitesSorting } = useSitesSorting();
 	const sortedSites = useSitesListSorting( allSites, sitesSorting );
 
 	// Call the generateCommandsArray function to get the commands array
-	let commands = useCommandsArrayWpcom( { setSelectedCommandName } );
+	const commands = useCommandsArrayWpcom( {
+		setSelectedCommandName,
+	} );
 
-	if ( 'function' === typeof filter ) {
-		commands = commands.filter( filter );
-	}
+	const currentPath = useSelector( ( state: object ) => getCurrentRoute( state ) );
+
+	const sortedCommands = commands
+		.filter( ( command ) => {
+			// Exclude "viewMySites" command when the current path is /sites
+			const isViewMySites = command.name === 'viewMySites';
+			return ! ( isViewMySites && currentPath === '/sites' );
+		} )
+		.sort( ( a, b ) => {
+			// Check if the current command is "viewMySites"
+			const isViewMySitesWithContextual = a.name === 'viewMySites';
```

Comment:
I could not find a good name for these - open to better or more readable suggestions